lib/sycamore/sycamore/writers/duckdb_writer.py

```diff
@@ -1,70 +1,116 @@
+    pass
 
 
 @dataclass
 class DuckDBTargetParams(BaseDBWriter.TargetParams):
-    table_name: str
+    parquet_location: str
 
     def compatible_with(self, other: BaseDBWriter.TargetParams) -> bool:
         if not isinstance(other, DuckDBTargetParams):
             return False
-        if self.table_name != other.table_name:
+        if self.parquet_location != other.parquet_location:
             return False
         return True
 
 
 class DuckDBClient(BaseDBWriter.Client):
     def __init__(self, client_params: DuckDBClientParams):
-        dict_params = asdict(client_params)
-        db_name = dict_params["db_name"] if "db_name" in dict_params else ":memory:"
-        self._client = duckdb.connect(db_name)
+        pass
 
     @classmethod
     def from_client_params(cls, params: BaseDBWriter.ClientParams) -> "DuckDBClient":
         assert isinstance(params, DuckDBClientParams)
         return DuckDBClient(params)
 
-    def write_many_records(self, records: list[BaseDBWriter.Record], target_params: BaseDBWriter.TargetParams):
-        assert isinstance(target_params, DuckDBTargetParams)
+    def write_many_records(
+        self, records: list[BaseDBWriter.Record], target_params: BaseDBWriter.TargetParams, batch_size: int = 1000
+    ):
+        N = 10
+        dict_params = (
+            asdict(target_params)
+            if is_dataclass(target_params)
+            else (target_params.__dict__ if hasattr(target_params, "__dict__") else {})
+        )
         assert _narrow_list_of_doc_records(records), f"Found a bad record in {records}"
-        with self._client:
-            for r in records:
-                self._client.execute(
-                    f"""INSERT INTO {target_params.table_name}
-                                     VALUES ({r.uuid}, {r.properties}, {r.embeddings})"""
-                )
+        assert isinstance(target_params, DuckDBTargetParams)
+
+        # Determine file location
+        parquet_location = dict_params.get("parquet_location", "./tmp/duckdb")
+        file_location = os.path.join(parquet_location, "".join(random.choices(string.ascii_uppercase, k=N)) + ".csv")
+        while os.path.isfile(file_location):
+            file_location = os.path.join(
+                parquet_location, "".join(random.choices(string.ascii_uppercase, k=N)) + ".csv"
+            )
+        # Prepare the CSV file and directory
+        directory = os.path.dirname(file_location)
+        os.makedirs(directory, exist_ok=True)
+
+        headers = ["uuid", "embeddings", "properties", "text_representation", "bbox", "shingles", "type"]
+
+        def write_batch(batch_data: dict):
+            # If the file doesn't exist, write headers first
+            file_exists = os.path.isfile(file_location)
+
+            with open(file_location, "a", newline="") as csvfile:
+                writer = csv.DictWriter(csvfile, fieldnames=headers)
+                if not file_exists:
+                    writer.writeheader()
```

Comment:
potential race condition in this logic. maybe drop headers altogether and assume them in the DB table creation phase?